Update mob_suite to 2.0.5

dfornika opened the pull request:
- Update `mob_recon` and `mob_typer` tools to use mob-suite v2.0.2

dfornika commented:
I've had some trouble getting a working tool wrapper for `mob_hostrange`, so I've removed that tool from this pull request. Can I request that this PR be merged without the `mob_hostrange` tool? I'd be happy to add my work-in-progress `mob_hostrange` tool back in another PR.

kbessonov1984 commented:
It is perfectly fine to skip mob_hostrange tool as it is mostly used to query the reference database by replicon, relaxase, plasmid cluster  The mob_typer invokes full host range report internally. I see no need to have mob_hostrange module in Galaxy.

dfornika commented:
@Takadonet Could I request a review? (I'm not able to use the integrated GitHub review request feature on this repo)

dfornika commented:
@kbessonov1984 could you review this PR please? Besides updating the version number, the main change that I made was to provide the chromosomal assembly and the collection of plasmid assemblies as separate outputs from `mob_recon`.

bgruening commented on tools/mob_suite/mob_typer.xml:
```diff
@@ -16,7 +16,7 @@
    --min_ori_ident '$adv_param.min_ori_ident'
    --min_mpf_ident '$adv_param.min_mpf_ident'
-    --outdir '.';
+    --outdir 'outdir';
```
The `ln` above should be single quoted and ; should be replaced by &&
Actually all double quotes should be replaced by a single quote.

kbessonov1984 commented:
@dfornika Great work! I am testing this Galaxy recipe in my docker container and will get back shortly. I would like to enhance this wrapper with the additional host range output (ascii tree, and reports).

I see that all plasmid sequences are moved to a separate folder `plasmids` and separate data collection for all plasmid sequences is prepared in the history to make it easier to download all of them as a collection. In the same vein, would it be better to rename chromosome to `chromosomal sequences (chromosome.fasta)` on line 71 of `mob_recon.xml` so that user will understand that these sequences are predicted to be chromosomal. 

Also for the `mob_typer` I would add additional outputs including the host range tree (both ascii and newick formats) and host range NCBI and literature reports, not only `plasmid report`.

kbessonov1984 commented on tools/mob_suite/mob_recon.xml:
```diff
@@ -1,25 +1,25 @@
-<tool id="mob_recon" name="MOB-Recon" version="1.4.9.1+galaxy0">
+<tool id="mob_recon" name="MOB-Recon" version="2.0.2+galaxy0">
```
Perhaps update to version 2.0.5 as it brings the same features of the version 2.0.2 but solves some host range issues linked to non-ascii chars and mean transfer rate calculation?

kbessonov1984 commented on tools/mob_suite/mob_recon.xml:
```diff
@@ -1,25 +1,25 @@
   <description>Type contigs and extract plasmid sequences</description>
   <requirements>
-     <requirement type="package" version="1.4.9.1">mob_suite</requirement>
+     <requirement type="package" version="2.0.2">mob_suite</requirement>
```
Update to 2.0.5?

kbessonov1984 commented on tools/mob_suite/mob_typer.xml:
```diff
@@ -1,12 +1,12 @@
   <description>Get the plasmid type and mobility given its sequence</description>
   <requirements>
-     <requirement type="package" version="1.4.9.1">mob_suite</requirement>
+     <requirement type="package" version="2.0.2">mob_suite</requirement>
```
Update to 2.0.5?

kbessonov1984 commented on tools/mob_suite/mob_typer.xml:
```diff
@@ -1,12 +1,12 @@
-<tool id="mob_typer" name="MOB-Typer" version="1.4.9.1+galaxy0">
+<tool id="mob_typer" name="MOB-Typer" version="2.0.2+galaxy0">
```
Update to 2.0.5?

dfornika commented:
Can I suggest that we keep the RefSeq and Literature trees as separate outputs, instead of merging them together into collections? I've found that it's difficult to pick specific outputs out of a collection when designing a pipeline (that's why I moved the `chromosome.fasta` out into a separate output).

I've added the ASCII and newick tree outputs. I should be able to add the remaining outputs fairly soon.

kbessonov1984 commented:
Hi Dan, I was also working on your commit trying to solve issue of empty Newick files. I finally managed to do it with `<discover_datasets>` tag, but one needs to set `assign_primary_output` to `true` i.e. `<discover_datasets pattern=".+_refseqhostrange_phylogeny_tree\.nwk" visible="true" format="newick" assign_primary_output="true"/>`. Otherwise it was giving me an empty file and I was thinking that there is an issue with the regex. 

I like the idea of replacing the <discover_datasets> tag for simpler <data> tags with the wild card `*`. It is very elegant and simple

I am also OK with keeping output data from the host range in the separate files rather than in a collection. In the future it will be nice to put all host range reports in a collection dataset to prevent cluttering of the history.

I also suggest to change from `${tool.name} on ${on_string}` to `${tool.name} on ${input.element_identifier}` in all instances as this would allow to refer to input file name rather than generic `data 1`. 

Otherwise I am OK with this implementation. It works great on my end.

dfornika commented:
@Takadonet can we merge?

Takadonet commented:
For sure. Does not have to go thru me to be merged. Anyone that reviews PR
should merge it.

On Wed., Dec. 11, 2019, 16:16 Dan Fornika, <notifications@github.com> wrote:

> @Takadonet <https://github.com/Takadonet> can we merge?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/phac-nml/galaxy_tools/pull/160?email_source=notifications&email_token=AAAPZXGPB73RSVVO6D6PC2DQYFRDHA5CNFSM4JNC2GG2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEGUYM4A#issuecomment-564758128>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAPZXHDJOPQE3HT3J2UUSDQYFRDHANCNFSM4JNC2GGQ>
> .
>
